app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -84,6 +87,36 @@ class HomeFragment : Fragment() {
         settingsButton.setOnClickListener { v ->
             onSettingsPressed?.invoke()
         }
+
+        registerForContextMenu(view)
+    }
+
+    override fun onCreateContextMenu(menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?) {
+        super.onCreateContextMenu(menu, v, menuInfo)
+        activity.menuInflater.inflate(R.menu.menu_context_hometile, menu)
+    }
+
+    override fun onContextItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.remove -> {
+                val homeTileAdapter = tileContainer.adapter as HomeTileAdapter
+                val tileToRemove = homeTileAdapter.getItemAtPosition(getFocusedTilePosition())
+                // This assumes that since we're deleting from a Home Tile object that we created
+                // that the Uri is valid, so we do not do error handling here.
+                if (tileToRemove is BundledHomeTile) {
```

Comment:
I'm seeing a compile error here for some reason, it looks like maybe because this is in an `else` the tileToRemove.url isn't guaranteed to not be null?

You can make this a `when` case, since HomeTile is a sealed class (there are a fixed number of cases), so like:

```
when (tileToRemove) {
    is BundledHomeTile -> {...}
    is CustomHomeTile -> {...}
}